### ISSUE TITLE
Can't use AMP components on regular Next.js page

### ISSUE THREAD

**Issue by iliran11** [zeit/next.js#8548]
Continuing this bug report: https://github.com/zeit/next.js/issues/8542

I am using:
* next.js - 9.0.5

I need to use AMP components on the regular AMP page too. (continue using the components I already used and build on top of it more features).

I'm writing this code for a page and visit the non-amp page:

```
import Head from 'next/head';
import { useAmp } from 'next/amp'


export const config = {
  hybrid: true
};

export default () => {
  const isAmp = useAmp()
    return (
      <div>
        {!isAmp && <Head>
          <script async src="https://cdn.ampproject.org/v0.js"></script>
        </Head>}
        <p>hi</p>
      </div>
    );
  };  
```
When i'm typing this in the chrome console on the non-AMP page:
`customElements.get('amp-img')`

I get this response, like it already exists somehow: 
`ƒ c(a){return d.call(this,a)||this}`

anyway, I get this error while trying to load the page: 

```
Uncaught (in promise) DOMException: Failed to execute 'define' on 'CustomElementRegistry': the name "amp-img" has already been used with this registry
    at y.Xc.y.define (https://cdn.ampproject.org/v0.js:67:7)
    at y.Xc.y.define (https://cdn.ampproject.org/v0.js:67:7)
    at Jj (https://cdn.ampproject.org/v0.js:288:769)
    at vj.Cd (https://cdn.ampproject.org/v0.js:565:40)
    at uj (https://cdn.ampproject.org/v0.js:247:834)
    at qj.vf (https://cdn.ampproject.org/v0.js:251:112)
    at https://cdn.ampproject.org/v0.js:251:319
```

And I can't use AMP components.
Please help.

**Comment by timneutkens** [comment]
I closed the issue as per the issue template:

```
# Question about Next.js

GitHub Issues are reserved for Bug reports and Feature requests. The best place to get your question answered is to post it on https://spectrum.chat/next-js.

```

**Comment by iliran11** [comment]
@timneutkens I  think it's a bug.

**Comment by timneutkens** [comment]
@iliran11 not really, you're doing something non-standard in your own custom code.

**Comment by iliran11** [comment]
@timneutkens 
It is standard.

I've been in a google workshop and they said it's a best practice to use AMP components even in a non-valid amp page.
it speeds up development. especially if I already developed AMP page with AMP components using your mechanism.

So at least mention in the docs that it is not possible to use AMP components with hybrid configuration (when your request the non-valid AMP page).

It basically means I need to maintain 2 sets of components (AMP and non-AMP) and could make you think twice before starting AMP page with next.js

**Comment by sebastianbenz** [comment]
@iliran11 what's your goal?

**1. Create an AMP page that requires additional non-supported functionality (e.g. ad formats, payments, ...)?** 

This is indeed a case in which we recommend to break AMP validity (if the pages are not search relevant). However, AMP still should be used as the main client-side framework.

Sidenote: a slightly related problem is that it's currently not possible to use experimental components which are not yet supported by the AMP validator. For this case, it'd be nice to have an option to disable AMP validation.

**2. Use AMP components on your existing React-based pages?** 

This is not really supported due to potential AMP and React runtime conflicts. However, this is a case we're actively working on adding future support: 

> To then make it possible to re-use existing templates you’ve built with AMP components, we’re working on what we call Bento AMP, the ability to use AMP components in an “un-managed” way, without loading AMP’s main JS file (v0.js), and coexisting with other web components and frameworks on the same page.

https://blog.amp.dev/2019/05/01/amp-as-your-web-framework/

**Comment by iliran11** [comment]
@sebastianbenz my goal is number #2...
thanks for this clear answer.

1. is there a way I can track the estimated date of release?
2. Are you going to validate on next.js that bento AMP is going to work on regular next.js page as part of the develop process?

**Comment by sebastianbenz** [comment]
re 1: Unfortunately, we haven't got a clear release date yet for Bento app as there still are a few technical challenges we have to solve first. The best place to watch for updates is the [AMP blog](https://blog.amp.dev). 

re 2: As part of the work on Bento AMP, we'll make sure that components will be 100% compatible with React.

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
